extensions/modules/mail/src/main/java/org/exist/xquery/modules/mail/MessageListFunctions.java

```diff
@@ -557,227 +544,166 @@ private SearchTerm[] parseChildSearchTerms( Node terms ) throws XPathException
+		if (pattern.isEmpty()) {
+			throw new XPathException(this, "Pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) );
 		}
-		
-		return( st );
+		return new SubjectTerm( pattern );
 	}
 
-	private SearchTerm parseBodyTerm( Node terms ) throws XPathException
-	{
-		SearchTerm	st = null;
+	private SearchTerm parseBodyTerm( Node terms ) throws XPathException {
+		final String pattern = ((Element)terms).getAttribute("pattern");
 		
-		String pattern  = ((Element)terms).getAttribute( "pattern" );
-		
-		if( pattern != null && !pattern.isEmpty()) {
-			st = new BodyTerm( pattern );
-		} else {
-			throw( new XPathException(this, "Pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+		if (pattern.isEmpty()) {
+			throw new XPathException(this, "Pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ));
 		}
-		
-		return( st );
+		return new BodyTerm(pattern);
 	}
 
-	private SearchTerm parseRecipientTerm( Node terms ) throws XPathException
-	{
+	private SearchTerm parseRecipientTerm( Node terms ) throws XPathException {
 		SearchTerm	st = null;
 		
 		String pattern  = ((Element)terms).getAttribute( "pattern" );
 		String type     = ((Element)terms).getAttribute( "recipientType" );
-		
-		if( StringUtils.isEmpty(type) ) {
-			throw( new XPathException(this, "recipientType not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+		String _type     = ((Element)terms).getAttribute( "type" );
+
+		if (type.isEmpty()) {
+			throw new XPathException(this, "recipientType not specified for term with type: " + _type);
 		}
 		
-		if( pattern != null && !pattern.isEmpty()) {
-			Message.RecipientType rtype = null;
-			
-			if( "to".equalsIgnoreCase(type) ) {
-				rtype = Message.RecipientType.TO;
-			} else if( "cc".equalsIgnoreCase(type) ) {
-				rtype = Message.RecipientType.CC;
-			} else if( "bcc".equalsIgnoreCase(type) ) {
-				rtype = Message.RecipientType.BCC;
-			} else {
-				throw( new XPathException(this, "Invalid recipientType: " + type + ", for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
-			}
-			
-			st = new RecipientStringTerm( rtype, pattern );
+		if (pattern.isEmpty()) {
+			throw new XPathException(this, "Pattern attribute must be specified for term with type: " + _type);
+		}
+
+		Message.RecipientType rtype = null;
+		if ("to".equalsIgnoreCase(type) ) {
+			rtype = Message.RecipientType.TO;
+		} else if( "cc".equalsIgnoreCase(type) ) {
+			rtype = Message.RecipientType.CC;
+		} else if( "bcc".equalsIgnoreCase(type) ) {
+			rtype = Message.RecipientType.BCC;
 		} else {
-			throw( new XPathException(this, "Pattern attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+			throw new XPathException(this, "Invalid recipientType: " + type + ", for term with type: " + ((Element)terms).getAttribute("type"));
 		}
```

Comment:
Here also a switch as function could be used to simplify the logic...

Reply:
I propose to change this in a followup PR